appsTransfer/src/main/java/org/kinecosystem/appstransfer/view/customview/AppsTransferList.kt

```diff
@@ -129,28 +126,30 @@ private class AppsTransferAdapter(private val context: Context) : RecyclerView.A
             appName.text = app.name
             icon.load(app.iconUrl)
             app.identifier?.let {
-                if (context.isAppInstalled(it)) {
-                    if (app.canTransferKin) {
-                        actionText.background = ContextCompat.getDrawable(context, R.drawable.kin_button_rounded_drawable)
-                        actionText.text = context.getString(R.string.apps_transfer_kin)
-                        actionText.setTextColor(ContextCompat.getColor(context, R.color.kin_transfer_btn_text))
-                        actionText.visibility = View.VISIBLE
-                        status.visibility = View.GONE
-                        appName.gravity = Gravity.CENTER_VERTICAL
+                if (app.canSendAndReceiveKin()) {
+                    if (context.isAppInstalled(it)) {
+                        if (app.canReceiveKin) {
+                            actionText.background = ContextCompat.getDrawable(context, R.drawable.kin_button_rounded_drawable)
+                            actionText.text = context.getString(R.string.apps_transfer_kin)
+                            actionText.setTextColor(ContextCompat.getColor(context, R.color.kin_transfer_btn_text))
+                            actionText.visibility = View.VISIBLE
+                            status.visibility = View.GONE
+                            appName.gravity = Gravity.CENTER_VERTICAL
```

Comment:
you could refactor to a method called with different values for each of the 3 cases